File: src/WellTool.Extra/hutool-extra/src/main/java/cn/hutool/extra/expression/engine/qlexpress/QLExpressEngine.java
package cn.hutool.extra.expression.engine.qlexpress;

import cn.hutool.extra.expression.ExpressionEngine;
import cn.hutool.extra.expression.ExpressionException;
import com.ql.util.express.DefaultContext;
import com.ql.util.express.ExpressRunner;
import com.ql.util.express.config.QLExpressRunStrategy;

import javax.naming.InitialContext;
import java.lang.reflect.Method;
import java.util.Collection;
import java.util.Map;

/**
 * QLExpress引擎封装<br>
 * 见：https://github.com/alibaba/QLExpress
 *
 * @author looly
 * @since 5.8.9
 */
public class QLExpressEngine implements ExpressionEngine {

	private final ExpressRunner engine;

	/**
	 * 构造
	 */
	public QLExpressEngine() {
		engine = new ExpressRunner();

		// issue#3994@Github
		// Enforce blacklisting of high-risk method invocations
		QLExpressRunStrategy.setForbidInvokeSecurityRiskMethods(true);
		// Explicitly forbid JNDI lookup calls through InitialContext
		QLExpressRunStrategy.addSecurityRiskMethod(InitialContext.class, "doLookup");
	}

	@Override
	public Object eval(final String expression, final Map<String, Object> context, Collection<Class<?>> allowClassSet) {
		// issue#3994@Github
		if (null != allowClassSet) {
			for (Class<?> clazz : allowClassSet) {
				for (Method method : clazz.getDeclaredMethods()) {
					QLExpressRunStrategy.addSecureMethod(clazz, method.getName());
				}
			}
		}
		final DefaultContext<String, Object> defaultContext = new DefaultContext<>();
		defaultContext.putAll(context);
		try {
			return engine.execute(expression, defaultContext, null, true, false);
		} catch (final Exception e) {
			throw new ExpressionException(e);
		}
	}
}

File: src/WellTool.Extra/Expression/Engine/QlExpress/QlExpressEngine.cs
using System;
using System.Collections.Generic;

namespace WellTool.Extra.Expression.Engine.QlExpress
{
    /// <summary>
    /// QLExpress表达式引擎实现
    /// 
    /// 需要安装 QLExpress 或类似 NuGet 包
    /// 项目地址：https://github.com/alibaba/QLExpress
    /// </summary>
    public class QlExpressEngine : ExpressionEngine
    {
        /// <summary>
        /// 构造
        /// </summary>
        public QlExpressEngine()
        {
        }

        /// <summary>
        /// 执行表达式
        /// </summary>
        /// <param name="expression">表达式</param>
        /// <param name="env">环境变量</param>
        /// <returns>执行结果</returns>
        public override object Eval(string expression, IDictionary<string, object> env)
        {
            // TODO: 需要集成 QLExpress.NET 或类似库
            return null;
        }
    }
}
